website/project/model.py

```diff
@@ -852,15 +855,43 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
         if inst.auth(user):
             self.primary_institution = inst
+            self.affiliated_institutions.append(inst)
             self.save()
+            self.add_log(
+                action=NodeLog.INSTITUTION_ADDED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
             return True
         return False
```

Comment:
Reviewing this led me to investigate this `User#has_inst_auth`  (https://github.com/CenterForOpenScience/osf.io/blob/8fe462a4ab2b9bb6885e0326547566e20e66f827/framework/auth/core.py#L1359-L1362), for which I have a couple minor suggestions:

- Avoid unnecessary abbreviations and keep naming consistent. You have `remove_institution` and `has_inst_auth`.
- The logic can be simplified to `return inst in self.affiliated_institutions`
- Is `has_inst_auth` an accurate name? It looks like it's just checking affiliation. Maybe `is_affiliated_with_instutiton(inst)`?